robocorp-code/src/robocorp_code/rcc.py

```diff
@@ -324,33 +326,25 @@ def _run_rcc(
         )
         return ActionResult(success=True, message=None, result=output)
 
-    _TEMPLATES = {
-        "standard": "Standard - Robot Framework Robot.",
-        "python": "Python - Python Robot.",
-        "extended": "Extended - Robot Framework Robot with additional scaffolding.",
-    }
-
     @implements(IRcc.get_template_names)
-    def get_template_names(self) -> ActionResult[List[str]]:
-        result = self._run_rcc("robot initialize -l".split())
+    def get_template_names(self) -> ActionResult[List[RobotTemplate]]:
+        result = self._run_rcc("robot initialize -l --json".split())
         if not result.success:
             return ActionResult(success=False, message=result.message)
 
-        output = result.result
+        output: Dict[str, str] = []
+        try:
+            output = json.loads(result.result)
+        except json.decoder.JSONDecodeError as e:
+            return ActionResult(success=False, message=f"Invalid output format: {e}")
         if output is None:
```

Comment:
Instead of checking `output is None`, this should check if `result.result is None` (this should be done before trying to load the json from it).

Reply:
Good catch! I pushed the fix in a separate commit, let me know if you want me to squash the commits, if you are not merging with squash.